src/rgw/rgw_op.cc

```diff
@@ -405,7 +406,11 @@ static int get_multipart_info(struct req_state *s,
   meta_obj = s->bucket->get_object(rgw_obj_key(meta_oid, string(), mp_ns));
   meta_obj->set_in_extra_data(true);
 
-  return get_multipart_info(s, meta_obj.get(), upload_info);
+  int ret = get_multipart_info(s, meta_obj.get(), upload_info);
+  if (ret >= 0 && attrs) {
+    *attrs = meta_obj.get()->get_attrs();
```

Comment:
You don't need the .get() to dereference a unique_ptr, meta_obj->get_attrs() works fine.

Reply:
done